item_purchase_generic.lua

```diff
@@ -0,0 +1,17 @@
+local bot = GetBot()
+
+if bot:IsInvulnerable() or bot:IsHero() == false or bot:IsIllusion()
+then
+	return
+end
+
+local HeroInfoFile = "NOT IMPLEMENTED"
+
+if bot:IsHero() then
+	HeroInfoFile = require(GetScriptDirectory() .. "/hero/" .. string.gsub(GetBot():GetUnitName(), "npc_dota_hero_", ""));
+end
+
+-- TODO: Call ItemPurchaseSystem directly
+function ItemPurchaseThink()
```

Comment:
You should already be able to just call `ItemPurchaseSystem.BuySupportItem()` and 	`ItemPurchaseSystem:ItemPurchaseExtend()` directly.  `CreateItemInformationTable` adds the `itemInformationTable` to the bot itself, so it's always there when you use `GetBot()`.  

I think if you want to be _extremely_ direct you can do
```
local ItemPurchaseSystem = dofile(GetScriptDirectory() .. "/util/ItemPurchaseSystem")
local ItemPurchaseThink = ItemPurchaseSystem:ItemPurchaseExtend
```
You'd have to incorporate / call `BuySupportItem()` in `ItemPurchaseExtend()`, but that probably makes sense to do anyway.

Reply:
Thanks a lot! After your explanation I finally understand what the CreateItemInformationTable function actually does.

Regarding `BuySupportItem()`, I think I will use RoleUtility to judge whether the hero is a support.